packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -4056,3 +4058,177 @@ export class ITxna extends Op {
+ * pops from stack: [...stack, uint64]
+ * push to stack [...stack, value of field]
+ */
+export class Gtxnas extends Gtxna {
+  /**
+   * Sets `field`(Transaction Field) and
+   * `txIdx`(Transaction Group Index) values according to arguments passed.
+   * @param args Expected arguments: [transaction group index, transaction field]
+   * // Note: Transaction field is expected as string instead of number.
+   * For ex: `Fee` is expected and `0` is not expected.
+   * @param line line number in TEAL file
+   * @param interpreter interpreter object
+   */
+  constructor (args: string[], line: number, interpreter: Interpreter) {
+    // NOTE: 100 is a mock value.
+    // In txnas & gtxnas opcodes, index is fetched from top of stack.
+    // eg. [ int 1; gtxnas 0 Accounts; ], [ gtxna 0 Accounts 1].
+    super([...args, "100"], line, interpreter);
+    assertLen(args.length, 2, line);
+  }
+
+  execute (stack: TEALStack): void {
+    this.assertMinStackLen(stack, 1, this.line);
+    const top = this.assertBigInt(stack.pop(), this.line);
+    this.idx = Number(top);
+    super.execute(stack);
+  }
+}
+
+/**
+ * gtxnsas F:
+ * pop an index A and an index B. push Bth value of the array
+ * field F from the Ath transaction in the current group
+ * pops from stack: [...stack, {uint64 A}, {uint64 B}]
+ * push to stack [...stack, value of field]
+ */
+export class Gtxnsas extends Gtxna {
+  /**
+   * Sets `field`(Transaction Field)
+   * @param args Expected arguments: [transaction field]
+   * // Note: Transaction field is expected as string instead of number.
+   * For ex: `Fee` is expected and `0` is not expected.
+   * @param line line number in TEAL file
+   * @param interpreter interpreter object
+   */
+  constructor (args: string[], line: number, interpreter: Interpreter) {
+    // NOTE: 100 is a mock value.
+    // In gtxnsas opcode, {tx-index, index of array field} is fetched from top of stack.
+    // eg. [ int 0; int 1; gtxnsas Accounts; ], [ gtxna 0 Accounts 1].
+    super(["100", args[0], "100"], line, interpreter);
+    assertLen(args.length, 1, line);
+  }
+
+  execute (stack: TEALStack): void {
+    this.assertMinStackLen(stack, 2, this.line);
+    const arrFieldIdx = this.assertBigInt(stack.pop(), this.line);
+    const txIdxInGrp = this.assertBigInt(stack.pop(), this.line);
+    this.idx = Number(arrFieldIdx);
+    this.txIdx = Number(txIdxInGrp);
+    super.execute(stack);
+  }
+}
+
+// pushes Arg[N] from LogicSig argument array to stack
+// Pops: ... stack, uint64
+// push to stack [...stack, bytes]
+export class Args extends Arg {
+  /**
+   * Gets the argument value from interpreter.args array.
+   * store the value in _arg variable
+   * @param args Expected arguments: none
+   * @param line line number in TEAL file
+   * @param interpreter interpreter object
+   */
+  constructor (args: string[], line: number, interpreter: Interpreter) {
+    super([...args, "100"], line, interpreter);
```

Comment:
why adding "100"?

Reply:
just a mock value, to inherit from super class. In this op the value corresponding to "100" will be fetched from stack